src/main/java/picard/sam/markduplicates/util/AbstractMarkDuplicatesCommandLineProgram.java

```diff
@@ -161,6 +161,10 @@
     protected void finalizeAndWriteMetrics(final LibraryIdGenerator libraryIdGenerator) {
         final Map<String, DuplicationMetrics> metricsByLibrary = libraryIdGenerator.getMetricsByLibraryMap();
         final Histogram<Short> opticalDuplicatesByLibraryId = libraryIdGenerator.getOpticalDuplicatesByLibraryIdMap();
+        final Histogram<Double> duplicateCountHist = libraryIdGenerator.getDuplicateCountHist();
```

Comment:
Why are these declared here? They aren't used until line 196 below. I think it would be clearer to remove the variables completely and inline the calls to the getters, e.g.
```
file.addHistogram(libraryIdGenerator.getDuplicateCountHist();
file.addHistogram(libraryIdGenerator.getOpticalDuplicateCountHist());
file.addHistogram(libraryIdGenerator.getNonOpticalDuplicateCountHist());
```